ledger-core/v2/virtual/object/object.go

```diff
@@ -202,3 +207,16 @@ func (sm *SMObject) stepReadyToWork(ctx smachine.ExecutionContext) smachine.Stat
 func (sm *SMObject) stepWaitIndefinitely(ctx smachine.ExecutionContext) smachine.StateUpdate {
 	return ctx.Sleep().ThenRepeat()
 }
+
+func (sm *SMObject) createWaitPendingOrderedSM(ctx smachine.ExecutionContext) {
+	syncSM := AwaitOrderedPendingSM{
+		sync: sm.MutableExecute,
+	}
+
+	// syncSM acquire MutableExecute semaphore in init step.
+	ctx.InitChildWithPostInit(func(ctx smachine.ConstructionContext) smachine.StateMachine {
+		return &syncSM
+	}, func() {
+		sm.AwaitPendingOrdered = &syncSM.stop
```

Comment:
BargeIn is by-value, so no need for ptr here

Reply:
Fixed